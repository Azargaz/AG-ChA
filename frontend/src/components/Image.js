import React from 'react'

import { makeStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';

const useStyles = makeStyles(() => ({
    center: {
        display: "block",
        marginLeft: "auto",
        marginRight: "auto"
    }
}));

function Image(props) {
    const classes = useStyles();
    const { src } = props;

    return (
        <Grid container justify="center" alignItems="center" spacing={3}>
            <Grid item xs={12}>
                <img src={src} alt="" height="250" className={classes.center}></img>
            </Grid>
        </Grid>
    )
}

export default Image
