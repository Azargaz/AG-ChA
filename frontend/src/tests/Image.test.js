import React from "react";
import { unmountComponentAtNode } from "react-dom";
import { create } from "react-test-renderer";

import Image from "../components/Image";

let container = null;

beforeEach(() => {
    // ustaw element DOM jako cel renderowania
    container = document.createElement("div");
    document.body.appendChild(container);
  });
  
  afterEach(() => {
    // posprzątaj po zakończeniu
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

    test("Matches the snapshot", () => {
      const button = create(<Image src={""} size={40} />);
      expect(button.toJSON()).toMatchSnapshot();
    });

