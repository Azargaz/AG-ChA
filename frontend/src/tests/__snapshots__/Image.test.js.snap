// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <img
      alt=""
      className="makeStyles-center-1"
      height={40}
      src=""
    />
  </div>
</div>
`;
